python/serve.py

```
@@ -98,29 +227,200 @@ def generate(
+    context = None
+    if not stateless:
+        conductor = nl_hecate.Conductor(
+            cfg.k,
+            list(cfg.chunk_sizes) if hasattr(cfg, "chunk_sizes") else [1] * cfg.k,
+        )
+        if gpu_model is None:
+            context = nl_hecate.ContextState(cfg.k, cfg.d_model)
+
+    mode_label = "stateless (full history)" if stateless else "stateful (CMS memory)"
+    print(f"\n{'─' * 60}")
+    print(f"  NL-Hecate Chat")
```

Ruff (0.15.1)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)